ci/test_narwhals.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+# Copyright (c) 2025, NVIDIA CORPORATION.
+
+# Support invoking test_python_cudf.sh outside the script directory
+cd "$(dirname "$(realpath "${BASH_SOURCE[0]}")")"/../ || exit
+
+# Common setup steps shared by Python test jobs
+source ./ci/test_python_common.sh test_python_narwhals
+
+RAPIDS_VERSION="$(rapids-version)"
+
+rapids-mamba-retry install \
+  --channel "${CPP_CHANNEL}" \
+  --channel "${PYTHON_CHANNEL}" \
+  "cudf-polars=${RAPIDS_VERSION}"
+
+rapids-logger "Check GPU usage"
+nvidia-smi
+rapids-print-env
+EXITCODE=0
+trap "EXITCODE=1" ERR
+set +e
+
+rapids-logger "pytest narwhals"
+git clone https://github.com/narwhals-dev/narwhals --depth=1
+pushd narwhals || exit
+pip install -U -e ".[dev]"
+
+rapids-logger "Check narwhals versions"
+python -c "import narwhals; print(narwhals.show_versions())"
+
+rapids-logger "Run narwhals tests for cuDF"
+python -m pytest \
+    --cache-clear \
+    --junitxml="${RAPIDS_TESTS_DIR}/junit-cudf-narwhals.xml" \
+    --numprocesses=8 \
```

Comment:
Like described in https://github.com/rapidsai/cudf/pull/18016#discussion_r1962365202, there is a `"${PARALLEL_LEVEL}"` environment variable that we can use here